File: sweet-shop-client/src/services/sweetService.js
import api from './api';

export const sweetService = {
  async getAllSweets() {
    const response = await api.get('/api/sweets');
    return response.data;
  },

  async searchSweets(params) {
    const response = await api.get('/api/sweets/search', { params });
    return response.data;
  },

  async addSweet(sweet) {
    const response = await api.post('/api/sweets', sweet);
    return response.data;
  },

  async updateSweet(id, sweet) {
    const response = await api.put(`/api/sweets/${id}`, sweet);
    return response.data;
  },

  async deleteSweet(id) {
    await api.delete(`/api/${id}`);
  },

  async purchaseSweet(id, qty = 1) {
    const response = await api.post(`/sweets/${id}/purchase`, null, { params: { qty } });
    return response.data;
  },

  async restockSweet(id, qty) {
    const response = await api.post(`/sweets/${id}/restock`, null, { params: { qty } });
    return response.data;
  },
};

File: sweet-shop-server/src/main/java/com/sweet_shop_server/sweet_shop_server/service/SweetService.java
package com.sweet_shop_server.sweet_shop_server.service;

import com.sweet_shop_server.sweet_shop_server.dto.SweetDTO;
import com.sweet_shop_server.sweet_shop_server.entity.Sweet;

import java.util.List;

public interface SweetService {

    // CREATE
    SweetDTO addSweet(SweetDTO sweetDTO);

    // READ
    List<SweetDTO> getAllSweets();

    List<SweetDTO> searchSweets(String name, String category, Double minPrice, Double maxPrice);

    SweetDTO getSweetById(Long id);

    // UPDATE
    SweetDTO updateSweet(Long id, SweetDTO sweetDTO);

    // DELETE
    void deleteSweet(Long id);  // Admin-only restriction handled in controller/security layer
}

